### ISSUE TITLE
parsing primary.txt

### ISSUE THREAD

**Issue by dlasher** [dgokcin/binance-triangle-arbitrage#2]
Any guidelines for how many entries can be in primary.txt? And/or how long each one has to be? 2, 3, 4 characters?

this file, as primary.txt will crash with the following error - "AttributeError: 'NoneType' object has no attribute 'encode'"

primary.txt:
---------
ADA
ALGO
ATOM
BAND
BAT
BCH
BITCNY
BNB
BNT
BSV
BTC
BTS
BUSD
COMP
CRO
DASH
DOT
EOS
ETC
ETH
JST
KNC
LEND
LINK
LTC
NEO
OKB
OMG
ONT
PAX
QTUM
SRM
SXP
TRX
TUSD
USDC
USDT
VET
WAVES
WETH
WGRT
XLM
XMR
XRP
XTZ
YFI
ZEC
ZIL
ZRX
----------

Thoughts?

**Comment by dlasher** [comment]
Remove XLM, and the list works... Strange?

**Comment by dgokcin** [comment]
@dlasher 
yes, definitely strange. I am using the same script in [this](https://gist.github.com/Valian/d16ef72a0e17ee82c0acf606d6a744d7) gist. I tweaked it to use a text file for the primary coins since I wanted to make them parametric. So I have no idea about the limitations of the text file.

This is a work in progress and any help is appreciated 🙂 .

**Comment by dlasher** [comment]
[primary.txt](https://github.com/dgokcin/binance-triangle-arbitrage/files/5158292/primary.txt)

I attached the primary.txt I'm using (top 250 coins by volume)

See if you can make that work?

**Comment by dgokcin** [comment]
@dlasher I can not reproduce your error, I can run the container successfully with this command:
```
 docker run -it -v "$(pwd)"/primary-2.txt:/app/primary.txt -e PUBLIC_KEY=<PUBLIC_KEY> -e SECRET_KEY=<SECRET_KEY> denizgokcin/binance-triangle-arbitrage:latest
```

Which OS are you using?
